Rdmp.Dicom/PipelineComponents/DicomSources/DicomFileCollectionSource.cs

```diff
@@ -210,37 +213,27 @@ private void ProcessDirectoryAsync(DataTable dt,DirectoryInfo directoryInfo, IDa
 
         private void ProcessDirectory(DataTable dt, DirectoryInfo directoryInfo,IDataLoadEventListener listener)
         {
-            listener.OnNotify(this, new NotifyEventArgs(ProgressEventType.Information, "Started Directory '" + directoryInfo.FullName + "' on Thread " + Thread.CurrentThread.ManagedThreadId));
-
-            FileInfo[] dicomFiles;
-            FileInfo[] zipFiles;
+            listener.OnNotify(this, new NotifyEventArgs(ProgressEventType.Information,
+                $"Started Directory '{directoryInfo.FullName}' on Thread {Thread.CurrentThread.ManagedThreadId}"));
 
-            try
+            //process all dcm files and archives in current directory
+            foreach (var file in directoryInfo.EnumerateFiles())
             {
-                dicomFiles = directoryInfo.EnumerateFiles().Where(f=>AmbiguousFilePath.IsDicomReference(f.FullName)).ToArray();
-                zipFiles = directoryInfo.EnumerateFiles("*.zip").ToArray();
-            }
-            catch (Exception e)
-            {
-                RecordError(directoryInfo.FullName,e);
-                return;
-            }
-
-            //process all dcm files in current directory
-            foreach (var dcmFile in dicomFiles)
                 try
                 {
-                    using (var fs = dcmFile.OpenRead())
-                        ProcessFile(fs, dt, dcmFile.FullName, listener);
+                    using var fs = file.OpenRead();
+                    if (!AmbiguousFilePath.IsDicomReference(file.FullName))
+                    {
+                        ProcessZipArchive(fs, dt, file.FullName, listener);
+                        continue;
+                    }
+                    ProcessFile(fs,dt,file.FullName,listener);
                 }
                 catch (Exception e)
                 {
-                    RecordError(dcmFile.FullName,e);
+                    RecordError(file.FullName, e);
                 }
-
-            foreach (var zipFile in zipFiles)
```

Comment:
I see that the iteration is now inside the try/catch.  Does that mean it will stop iterating the directory after the first file/zip in it has thrown?